model/infinispan/src/main/java/org/keycloak/cluster/infinispan/InfinispanClusterProviderFactory.java

```diff
@@ -195,6 +198,17 @@ public boolean isSupported(Config.Scope config) {
     @Listener
     public class ViewChangeListener {
 
+        @Merged
+        public void mergeEvent(MergeEvent event) {
+           localExecutor.execute(() -> {
+              var cm = workCache.getCacheManager();
+              cm.getAccessibleCacheNames().stream()
+                    .map(cm::getCache)
+                    .filter(c -> c.getCacheConfiguration().clustering().cacheMode() == CacheMode.LOCAL)
```

Comment:
Iterating over all local caches is good for a default configuration that Keycloak ships. 

Still I can see problems where people add their own additional caches to the Infinispan configuration.

Or they choose to switch a cache to local where it is by default distributed (for example, if they use sticky sessions and don't care a lot for their authentication sessions, they might configure it to be a local cache). 

So I'd rather iterate over all those caches which depend on the work cache's invalidations. There is a constant `InfinispanConnectionProvider.LOCAL_CACHE_NAMES` which contains those cache names. Despite its name, it should be read as "caches that a by default local where users shouldn't have a reason to configure them differently". And those implicitly depend on the work cache.